File: remoteApiBindings/python/python/getRobotLocation.py
import sim
import time

sim.simxFinish(-1)
clientID = sim.simxStart('127.0.0.1', 19999, True, True, 5000, 5)
if clientID != -1:
    print('已经连接到服务器')

    while True:

        a, hexa_base_handle = sim.simxGetObjectHandle(clientID, 'youBot', sim.simx_opmode_blocking)
        b, floor_visible_handle = sim.simxGetObjectHandle(clientID, 'ResizableFloor_5_25', sim.simx_opmode_blocking)
        c, pillar_visible_handle = sim.simxGetObjectHandle(clientID, '80cmHighPillar25cm', sim.simx_opmode_blocking)
        print(hexa_base_handle)
        print(floor_visible_handle)
        ret, positon = sim.simxGetObjectPosition(clientID, hexa_base_handle, floor_visible_handle, sim.simx_opmode_blocking)
        print(positon)
        if ret == sim.simx_return_ok:
            print("Position")
            print(positon)
        time.sleep(0.2)

File: remoteApiBindings/java/java/getRobotLocation.java
import coppelia.FloatWA;
import coppelia.IntW;
import coppelia.IntWA;
import coppelia.remoteApi;

public class getRobotLocation {
    public static void main(String[] args) {
        System.out.println("Program started");
        remoteApi sim = new remoteApi();
        sim.simxFinish(-1);
        int clientID = sim.simxStart("127.0.0.1", 19999, true, true, 5000, 5);
        if (clientID != -1) {
            System.out.println("已经连接到服务器");
            IntW floor_handel = new IntW(-1);
            IntW robot_handel = new IntW(-1);
            int floor_handel_ret = sim.simxGetObjectHandle(clientID, "ResizableFloor_5_25", floor_handel, sim.simx_opmode_blocking);
            int robot_handel_ret = sim.simxGetObjectHandle(clientID, "youBot", robot_handel, sim.simx_opmode_blocking);
            FloatWA position = new FloatWA(3);
            int ret = sim.simxGetObjectPosition(clientID, robot_handel.getValue(), floor_handel.getValue(), position, sim.simx_opmode_blocking);
            if (ret == 0) {
                System.out.println(String.format("Positon: {}", position));
                System.out.println(position);
            }
        }
    }
}
